
message Test_Request
{
    enum Request_Type {
        TEST_LSIC = 0;
        TEST_COMPARE = 1;
        TEST_ENC_COMPARE = 2;
        TEST_REV_ENC_COMPARE = 3;
        TEST_LINEAR_ENC_ARGMAX = 4;
        TEST_FHE = 5;
        TEST_CHANGE_ES = 6;

        DISCONNECT = 10;
    }
    required Request_Type type = 1;
    optional uint32 bit_size = 2;
    optional uint32 iterations = 3;
    optional bool use_lsic = 4;
    optional uint32 argmax_elements = 5;
}
